spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/SimpleStreamService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.service.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.dataflow.core.StreamDefinition;
+import org.springframework.cloud.dataflow.rest.util.DeploymentPropertiesUtils;
+import org.springframework.cloud.dataflow.server.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.StreamDeploymentRepository;
+import org.springframework.cloud.dataflow.server.stream.AppDeployerStreamDeployer;
+import org.springframework.cloud.dataflow.server.stream.StreamDeployers;
+import org.springframework.cloud.dataflow.server.stream.StreamDeploymentRequest;
+import org.springframework.cloud.deployer.spi.app.DeploymentState;
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.cloud.skipper.domain.PackageIdentifier;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import static org.springframework.cloud.dataflow.rest.SkipperStream.SKIPPER_KEY_PREFIX;
+
+/**
+ * {@link AppDeployerStreamDeployer} specific {@link AbstractStreamService}.
+ *
+ * @author Mark Pollack
+ * @author Ilayaperumal Gopinathan
+ * @author Christian Tzolov
+ */
+@Service
+public class SimpleStreamService extends AbstractStreamService {
```

Comment:
Can we call this call something like `AppDeployerStreamService`?

Reply:
I've been trying to avoid names like 'Classic'XXX. The AppDeployerXXX is good because will be consistent with the naming with some of the other non-skipper-path classes. 
My only concern with this name (here and in the other places) is that it is ambiguous given that the Skipper deployer is conceptually an app deployer as well. 
Still I will rename it to AppDeployerStreamService for now and we can think latter  fro better names for all AppDeployerXxxx names